### PR TITLE
5.15.2

### PR DESCRIPTION
**Pull request by LukasTy**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Create a new tag named with the release you just did `git tag -a v4.0.0-alpha.30 -m "Version 4.0.0-alpha.30" && git push upstream --tag`

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream master:docs-v5 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### Announce

- [ ] Follow the instructions in https://mui-org.notion.site/Releases-7490ef9581b4447ebdbf86b13164272d.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,48 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.16.0
+_Aug 11, 2022_
+
+We'd like to offer a big thanks to the 7 contributors who made this release possible. Here are some highlights ✨:
+
+- ✨ Improve quick filtering with row grouping (#5701) @alexfauquette
+- 📚 Documentation improvements
+- 🐞 Bugfixes
```
Comment (flaviendelangle) [inline CHANGELOG.md:13]: Did not noticed that before but sometime we write `Bug fixes` and sometime `Bugfixes`
I personally prefer the format but let's pick one and stick to it

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,48 @@
...
+- [DataGridPremium] Let quick filter search in row grouping children (#5701) @alexfauquette
+
+### `@mui/x-date-pickers@v5.0.0-beta.5` / `@mui/x-date-picker-pro@5.0.0-beta.5`
+
+#### Changes
+
+- [time picker] set clock focus outline to `none` (#5758) @LukasTy
+- [pickers] Fix theme augmentation with typescript (#5596) @alexfauquette
+- [pickers] Reset input value when locale is modified (#5310) @alexfauquette
+- [pickers] Support `disableHighlightToday` on MonthPicker and YearPicker (#5562) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:30]:
```suggestion
- [pickers] Support `disableHighlightToday` on `MonthPicker` and `YearPicker` (#5562) @flaviendelangle
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,48 @@
...
+- [pickers] fix to default to desktop mode when matchMedia is unavailable (#5684) @LukasTy
```
Comment (flaviendelangle) [inline CHANGELOG.md:31]:
```suggestion
- [pickers] Fallback to desktop mode when `matchMedia` is unavailable (#5684) @LukasTy
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,48 @@
...
+- [pickers] fix to trigger `onChange` when clearing or accepting `Invalid date` (#5740) @LukasTy
```
Comment (flaviendelangle) [inline CHANGELOG.md:32]:
```suggestion
- [pickers] Trigger `onChange` when clearing or accepting `Invalid date` (#5740) @LukasTy
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,48 @@
...
+
+### Docs
+
+- [docs] Add RFC GH issue template (#5739) @bytasv
+- [docs] Add description to the GridExportStateParams page (#5654) @oliviertassinari
```
Comment (flaviendelangle) [inline CHANGELOG.md:37]:
```suggestion
- [docs] Add description to the `GridExportStateParams` page (#5654) @oliviertassinari
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,48 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.16.0
```
Comment (alexfauquette) [inline CHANGELOG.md:6]: I thought it would be a patch. The quick filtering logic modification is kind of a bug fix

File: CHANGELOG.md (hunk 2 above)
Comment (alexfauquette) [inline CHANGELOG.md:30]:
```suggestion
- [pickers] Support `disableHighlightToday` on `MonthPicker` and `YearPicker` (#5562) @flaviendelangle
```

File: CHANGELOG.md (hunk 3 above)
Comment (alexfauquette) [inline CHANGELOG.md:31]:
```suggestion
- [pickers] fix to default mode to `'desktop'` when `matchMedia` is unavailable (#5684) @LukasTy
```

File: CHANGELOG.md (hunk 1 above)
Comment (LukasTy) [reply CHANGELOG.md:13]: 32 instances of `bugfixes` and 5 instances of `bug fixes`. 🤔 
I guess it's safe to say that we want to `stick` with `bugfixes`? 😄

File: CHANGELOG.md (hunk 6 above)
Comment (m4theushw) [inline CHANGELOG.md:6]:
```suggestion
## 5.16.0

```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,48 @@
...
+- [DataGrid] Catch error if rows freezing is not supported (#5711) @cherniavskii
+- [DataGrid] Preserve cell mode when entering edit mode while commiting (#5686) @m4theushw
+- [DataGridPremium] Let quick filter search in row grouping children (#5701) @alexfauquette
+
+### `@mui/x-date-pickers@v5.0.0-beta.5` / `@mui/x-date-picker-pro@5.0.0-beta.5`
+
+#### Changes
+
+- [time picker] set clock focus outline to `none` (#5758) @LukasTy
+- [pickers] Fix theme augmentation with typescript (#5596) @alexfauquette
```
Comment (m4theushw) [inline CHANGELOG.md:28]:
```suggestion
- [pickers] Fix theme augmentation with TypeScript (#5596) @alexfauquette
```

**Review by m4theushw — COMMENTED** [review]
Shouldn't it be a patch?

File: CHANGELOG.md (hunk 6 above)
Comment (LukasTy) [reply CHANGELOG.md:6]: I have to agree on that, even though there are certain changes in "API", but it's not related to `DataGrid`
For example: the `pickers` defaulting to `desktop` mode in tests, but those have their own `beta` version. 👍

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,48 @@
...
+
+- ✨ Improve quick filtering with row grouping (#5701) @alexfauquette
+- 📚 Documentation improvements
+- 🐞 Bugfixes
+
+### `@mui/x-data-grid@5.16.0` / `@mui/x-data-grid-pro@5.16.0` / `@mui/x-data-grid-premium@5.16.0`
+
+#### Changes
+
+- [DataGrid] Catch error if rows freezing is not supported (#5711) @cherniavskii
```
Comment (DanailH) [inline CHANGELOG.md:19]:
```suggestion
- [DataGrid] Catch errors if rows freezing is not supported (#5711) @cherniavskii
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,48 @@
...
+### Docs
+
+- [docs] Add RFC GH issue template (#5739) @bytasv
+- [docs] Add description to the GridExportStateParams page (#5654) @oliviertassinari
+- [docs] Improve the Events doc page (#5413) @flaviendelangle
+- [docs] Use new edit API in the introduction demos (#5728) @oliviertassinari
+
+### Core
+
+- [core] Remove duplicated FUNDING.yml file (#5656) @oliviertassinari
```
Comment (DanailH) [inline CHANGELOG.md:43]:
```suggestion
- [core] Remove duplicated `FUNDING.yml` file (#5656) @oliviertassinari
```

File: CHANGELOG.md (hunk 3 above)
Comment (flaviendelangle) [reply CHANGELOG.md:31]: https://github.com/mui/mui-x/pull/5761#discussion_r943409743

I prefer mine :D

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,49 @@
...
+
+#### Changes
+
+- [TimePicker] Set clock focus outline to `none` (#5758) @LukasTy
+- [pickers] Fix theme augmentation with TypeScript (#5596) @alexfauquette
+- [pickers] Reset input value when locale is modified (#5310) @alexfauquette
+- [pickers] Support `disableHighlightToday` on `MonthPicker` and `YearPicker` (#5562) @flaviendelangle
+- [pickers] Fallback to desktop mode when `matchMedia` is unavailable (#5684) @LukasTy
+- [pickers] Trigger `onChange` when clearing or accepting `Invalid date` (#5740) @LukasTy
+
```
Comment (cherniavskii) [inline CHANGELOG.md:34]:
```suggestion
- [pickers] Add `react-dom` to peerDependencies (#5752) @cherniavskii

```

Comment (LukasTy) [reply CHANGELOG.md:35]: I've added it as first change, because the script seems to list those in descending order. 😉

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,50 @@
...
+- [DataGrid] Catch errors if rows freezing is not supported (#5711) @cherniavskii
+- [DataGrid] Preserve cell mode when entering edit mode while commiting (#5686) @m4theushw
+- [DataGridPremium] Let quick filter search in row grouping children (#5701) @alexfauquette
+
+### `@mui/x-date-pickers@v5.0.0-beta.5` / `@mui/x-date-picker-pro@5.0.0-beta.5`
+
+#### Changes
+
+- [pickers] Add `react-dom` to peerDependencies (#5752) @cherniavskii
+- [TimePicker] Set clock focus outline to `none` (#5758) @LukasTy
```
Comment (alexfauquette) [inline CHANGELOG.md:29]: Only for esthetic reasons 

```suggestion
- [TimePicker] Set clock focus outline to `none` (#5758) @LukasTy
- [pickers] Add `react-dom` to peerDependencies (#5752) @cherniavskii
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md (hunk 11 above)
Comment (LukasTy) [reply CHANGELOG.md:29]: You are suggesting a change of order?
The order is in line with what would be produced by our script generating changelog. 🤔 
Are you sure we need this?

**Review by flaviendelangle — APPROVED** [review]

**Review by DanailH — APPROVED** [review]

**Pull request merged**